### ISSUE TITLE
[bug]: No Spacing in between Toasts

### ISSUE THREAD

**Issue by rampatra** [shadcn-ui/ui#5843]
### Describe the bug

https://github.com/user-attachments/assets/750fb4e5-5374-4934-a351-7f8d81b01803

No spacing when multiple Toasts are shown at once.

### Affected component/components

Toast

### How to reproduce

1. Just followed the instructions here - https://ui.shadcn.com/docs/components/toast
2. Increased the `TOAST_LIMIT` to 3
3. Saw the issue as per the video

### Codesandbox/StackBlitz link

https://github.com/user-attachments/assets/750fb4e5-5374-4934-a351-7f8d81b01803

### System Info

```bash
Brave Browser, macOS
```

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues

**Comment by DevFromSiberia** [comment]
I think the problem is pulling from radix-ui.

**Comment by cloudyonthemic** [comment]
Hi Ramptra,

You can resolve this issue by modifying the `toaster.tsx` component. Simply add a margin style to your code as shown in the example below:

```ts
'use client';

import { useToast } from '@/hooks/use-toast';
import {
 Toast,
 ToastClose,
 ToastDescription,
 ToastProvider,
 ToastTitle,
 ToastViewport,
} from '@/components/ui/toast';

export function Toaster() {
 const { toasts } = useToast();

 return (
  <ToastProvider>
   {toasts.map(function ({ id, title, description, action, ...props }) {
    return (
      // ADD MARGIN STYLE HERE I ADDED MARGIN BOTTOM OF 2
     <Toast key={id} {...props} className="mb-2">
      <div className="grid gap-1">
       {title && <ToastTitle>{title}</ToastTitle>}
       {description && <ToastDescription>{description}</ToastDescription>}
      </div>
      {action}
      <ToastClose />
     </Toast>
    );
   })}
   <ToastViewport />
  </ToastProvider>
 );
}
```

This should fix your problem of no spacing between toasts :)

**Comment by danwild** [comment]
It would be great if there was a way to do this without needing to directly modify the toaster component..

**Comment by cloudyonthemic** [comment]
I mean yeah but at the same time shouldn't be a big problem if you're familiar with web development, first thing you should check out is the component itself

Cheers 😄

**Comment by rampatra** [comment]
Just to clarify: the purpose of raising this bug here wasn't because I wasn't able to fix the issue in my code but for shadcn to include the fix in the main code/component itself. Maybe I should raise a PR and see whether shadcn accepts it as a solution.

**Comment by danwild** [comment]
> I mean yeah but at the same time shouldn't be a big problem if you're familiar with web development, first thing you should check out is the component itself
> 
> Cheers 😄

Just because you can, doesn't mean you _should_. 

I would expect an out-of-the-box component to display sensibly by default (which does not appear to be the case).

Failing this, it should provide a mechanism to customise the display (e.g. via props). Which currently, it also does not.

Thanks for raising the issue and PR @rampatra

**Comment by cloudyonthemic** [comment]
> > I mean yeah but at the same time shouldn't be a big problem if you're familiar with web development, first thing you should check out is the component itself
> > Cheers 😄
> 
> Just because you can, doesn't mean you _should_.
> 
> I would expect an out-of-the-box component to display sensibly by default (which does not appear to be the case).
> 
> Failing this, it should provide a mechanism to customise the display (e.g. via props). Which currently, it also does not.
> 
> Thanks for raising the issue and PR [@rampatra](https://github.com/rampatra)

You're absolutely right, I made a PR fixing it using props, let's hope for the best.

**Comment by shadcn** [comment]
This issue has been automatically marked as stale due to one year of inactivity. It will be closed in 7 days unless there’s further input. If you believe this issue is still relevant, please leave a comment or provide updated details. Thank you. (This is an automated message)

**Comment by shadcn** [comment]
This issue has been automatically closed due to one year of inactivity. If you’re still experiencing a similar problem or have additional details to share, please open a new issue following our current issue template. Your updated report helps us investigate and address concerns more efficiently. Thank you for your understanding! (This is an automated message)
